inspirehep/modules/migrator/tasks/records.py

```diff
@@ -285,17 +291,37 @@ def create_record(record):
 
 def record_insert_or_replace(json):
     """Insert or replace a record."""
+    def _store_texkey(uuid, record):
+        try:
+            inspire_texkey_minter(uuid, record)
+        except TexkeyMinterAlreadyValid:
+            logger.debug(
+                'The record with uuid {} has already a valid key'.format(
+                    str(uuid)
+                )
+            )
+        except TexkeyMinterError:
+            logger.error(
+                'Not able to produce texkey for the record with uuid: {}'.format(
+                    str(uuid)
+                )
+            )
+
     control_number = json.get('control_number', json.get('recid'))
     if control_number:
         pid_type = get_pid_type_from_schema(json['$schema'])
         try:
             pid = PersistentIdentifier.get(pid_type, control_number)
             record = InspireRecord.get_record(pid.object_uuid)
+            migrate_texkeys_from_legacy(pid.object_uuid, record)
```

Comment:
so what happens when there is an update from legacy, some texkeys had already been registered, others are new?

Reply:
As you can see here (https://github.com/inspirehep/inspire-next/pull/2526/files?diff=split#diff-19ddea958cb55c186c8bf6f2ff56d526R328) if the key has been already registered we do not store the key and we log the error.